master-firmware/src/strategy.cpp

```diff
@@ -548,6 +549,142 @@ struct BuildTowerLevel : actions::BuildTowerLevel {
     }
 };
 
+void strat_collect_wastewater(enum strat_color_t color)
+{
+    ballgun_tidy(&main_ballgun);
+    ballgun_deploy(&main_ballgun);
+    strategy_wait_ms(1000);
+
+    ballgun_charge(&main_ballgun);
+
+    trajectory_a_rel(&robot.traj, MIRROR(color, -95));
+    trajectory_wait_for_end(TRAJ_FLAGS_SHORT_DISTANCE /*TRAJ_FLAGS_ROTATION*/);
```

Comment:
Why not TRAJ_FLAGS_ROTATION?

Reply:
Also I think there should be some error handling in there. Like if you are blocked you should maybe move away from the dispenser then try to retract. Otherwise you might break something.